### ISSUE TITLE
AWS build-ami playbook fails on write permissions

### ISSUE THREAD

**Issue by copejon** [openshift/openshift-ansible#10562]
#### Description

Following the [AWS Playbooks](https://github.com/openshift/openshift-ansible/tree/master/playbooks/aws), Step 1 Build AMI fails when writing to `root` owned dirs on remote hosts.

Attempted With
 - Branches: `release-3.11` and `master`, details below
 - OSes: Centos7 and Rhel 7.5 AWS marketplace AMIs

##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`

```
$ ansible --version
ansible 2.6.5
  config file = /home/jon/workspace/github.com/ansible-cloud-deploy/ansible.cfg
  configured module search path = [u'/home/jon/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

If you're operating from a **git clone**:

* The output of `git describe`

_release-3.11_
```
$ git describe
openshift-ansible-3.11.29-1-11-g912193b
```

_and_

_master_

```
$ git describe
openshift-ansible-4.0.0-0.32.0-2-gdc94ddd
```

##### Steps To Reproduce
1. Config

provisioning_vars.yml
```
openshift_aws_build_ami_ssh_user: centos # or ec2-user for rhel
openshift_deployment_type: origin
openshift_aws_region: us-east-1
openshift_clusterid: ocp  # example: example
openshift_cloudprovider_kind: aws
openshift_aws_ssh_key_name: aws  # example: myuser_key
openshift_aws_base_ami: ami-4bf3d731 # centos 
```

2. Run prerequisite playbook

`$ ansible-playbook -i playbooks/aws/provisioning-inventory.example.ini playbooks/aws/openshift-cluster/prerequisites.yml -e @provisioning_vars.yaml`

3. Run the build-ami playbook

`ansible-playbook -i playbooks/aws/provisioning-inventory.example.ini  playbooks/aws/openshift-cluster/build_ami.yml -e @provisioning_vars.yaml`

##### Expected Results

After step 3 above, I expected an OCP-ready AMI to be created.

##### Observed Results

The `Configure origin gpg keys` task fails to write the GPG keys citing a permission error, prematurely ending the playbook.

```
TASK [openshift_repos : Configure origin gpg keys] ***************************************************************************************************************************************************************************************************************************************************************************
Tuesday 30 October 2018  16:13:52 -0500 (0:00:00.075)       0:01:04.692 *******
fatal: [ec2-184-73-56-146.compute-1.amazonaws.com]: FAILED! => {"changed": false, "checksum": "e350bf139f91980ff9cc39877e14643ba3139da1", "msg": "Destination /etc/pki/rpm-gpg not writable"}
```

##### Additional Information

Workaround:
Create a _new_ base AMI after tweaking `visudo`

1. Start an instance of the base AMI
2. `sudo visudo`
3. Comment out the line 56
```     53 # Disable "ssh hostname sudo <cmd>", because it will show the password in clear.
     54 #         You have to run "ssh -t hostname sudo <cmd>".
     55 #
     56 #Defaults    requiretty <-- disable requiretty
```
4. Create an AMI from this instance
5. Edit `./roles/openshift_repos/tasks/centos_repos.yml` (is run for rhel and centos).  Add `become: true` to both tasks. A similar error occurs if not added to the second task.

```
---
# Note: OpenShift repositories under CentOS may be shipped through the
# "centos-release-openshift-origin" package which configures the repository.
# This task matches the file names provided by the package so that they are
# not installed twice in different files and remains idempotent.

- name: Configure origin gpg keys
  become: true # ADD 
  copy:
    src: "origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS"
    dest: "/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-PaaS"
  notify: refresh cache

# openshift_version is formatted to a standard string in openshift_version role.
# openshift_version is expected to be in format 'x.y.z...' here.
# Here, we drop the '.' characters and try to match the correct repo template
# for our corresponding openshift_version.
- name: Configure correct origin release repository
  become: true # ADD 
  template:
    src: "{{ item }}"
    dest: "/etc/yum.repos.d/{{ (item | basename | splitext)[0] }}"
  with_first_found:
    - "CentOS-OpenShift-Origin{{ ((openshift_version | default('')).split('.') | join(''))[0:2] }}.repo.j2"
    - "CentOS-OpenShift-Origin.repo.j2"
  notify: refresh cache
```

4. Rerun the build-ami playbook as before
5. _Aaaaand_ hit this error, which i'll open another issue for

```
TASK [Ensure openshift-ansible installer package deps are installed] *********************************************************************************************************************************************************************************************************************************************************
Tuesday 30 October 2018  13:15:38 -0500 (0:00:00.487)       0:01:07.460 *******
FAILED - RETRYING: Ensure openshift-ansible installer package deps are installed (3 retries left).
FAILED - RETRYING: Ensure openshift-ansible installer package deps are installed (2 retries left).
FAILED - RETRYING: Ensure openshift-ansible installer package deps are installed (1 retries left).
fatal: [ec2-34-228-208-86.compute-1.amazonaws.com]: FAILED! => {"attempts": 3, "changed": false, "msg": "Failure talking to yum: Could not contact any CDS load balancers: rhui2-cds01.us-east-1.aws.ce.redhat.com, rhui2-cds02.us-east-1.aws.ce.redhat.com."}
```

**Comment by skoussou** [comment]
On error no.5 have you found a solution? I have tried to root "become: yes"  but it still fails with unavailability of packages
playbooks/init/base_packages.yml

  - name: Ensure openshift-ansible installer package deps are installed
    become: yes
    package:
      name: "{{ pkg_list | join(',') }}"
      state: present
    vars:
      pkg_list_temp:
      - iproute
      - "{{ 'python3-dbus' if ansible_distribution == 'Fedora' else 'dbus-python' }}"
      - "{{ 'python3-PyYAML' if ansible_distribution == 'Fedora' else 'PyYAML' }}"
      - libsemanage-python
      - yum-utils
      - "{{ 'python3-docker' if ansible_distribution == 'Fedora' else 'python-docker-py' }}"
      pkg_list_non_fedora:
      - 'python-ipaddress'
      pkg_list_use_non_fedora: "{{ ansible_distribution != 'Fedora' | bool }}"
      pkg_list: "{{ pkg_list_non_fedora | ternary(pkg_list_non_fedora, []) + pkg_list_temp }}"
    register: result
    until: result is succeeded

**Comment by copejon** [comment]
@skoussou No, unfortunately.  I had to give up on using it.

**Comment by copejon** [comment]
Closing the issue do to old age.
